IndexedDB/idbobjectstore_getAll.html

```diff
@@ -0,0 +1,174 @@
+    req.onsuccess = onsuccess;
+    req.onerror = function() {
+      assert_unreached('open should not fail');
+    };
+    req.onupgradeneeded = function(evt) {
+      var connection = evt.target.result;
+
+      var store = connection.createObjectStore('generated',
+            {autoIncrement: true, keyPath: 'id'});
+      alphabet.forEach(function(letter) {
+        store.put({ch: letter});
+      });
+
+      store = connection.createObjectStore('out-of-line', null);
+      alphabet.forEach(function(letter) {
+        store.put('value-' + letter, letter);
+      });
+
+      store = connection.createObjectStore('empty', null);
+    };
+  };
+}
+
+function createGetAllRequest(t, storeName, connection, range, maxCount) {
+    var transaction = connection.transaction(storeName, 'readonly');
+    var store = transaction.objectStore(storeName);
+    // TODO(cmumford): Simplify once crbug.com/335871 is closed.
+    var req = maxCount !== undefined ? store.getAll(range, maxCount) :
+              range !== undefined ? store.getAll(range) : store.getAll();
+    req.onerror = t.unreached_func('getAll request should succeed');
+    return req;
+}
+
+doSetup(location.pathname + '-IDBObjectStore.getAll', 1, function(evt) {
+    var connection = evt.target.result;
+    async_test(function(t) {
+      var req = createGetAllRequest(t, 'out-of-line', connection, 'c');
+      req.onsuccess = t.step_func(function(evt) {
+          assert_array_equals(evt.target.result, ['value-c']);
+          t.done();
+      });
+    }, 'Single item get');
+
+    async_test(function(t) {
+      var req = createGetAllRequest(t, 'generated', connection, 3);
+      req.onsuccess = t.step_func(function(evt) {
+          var data = evt.target.result;
+          assert_true(Array.isArray(data));
+          assert_equals(data.length, 1);
+          assert_equals(data[0].id, 3);
+          assert_equals(data[0].ch, 'c');
+          t.done();
+      });
+    }, 'Single item get (generated key)');
+
+    async_test(function(t) {
+      var req = createGetAllRequest(t, 'empty', connection);
+      req.onsuccess = t.step_func(function(evt) {
+          assert_array_equals(evt.target.result, [],
+              'getAll() on empty object store should return an empty array');
+          t.done();
+      });
+    }, 'getAll on empty object store');
+
+    async_test(function(t) {
+      var req = createGetAllRequest(t, 'out-of-line', connection);
+      req.onsuccess = t.step_func(function(evt) {
+          assert_array_equals(evt.target.result,
+              alphabet.map(c => 'value-' + c));
+          t.done();
+      });
+    }, 'Get all values');
+
+    async_test(function(t) {
+      var req = createGetAllRequest(t, 'out-of-line', connection, undefined,
+                                    10);
+      req.onsuccess = t.step_func(function(evt) {
+          assert_array_equals(evt.target.result,
+              ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j']
```

Comment:
You don't "abcdef".split() anymore?  (I see no problem being explicit, just ask :P)